src/syntax-tree/ast.ts

```diff
@@ -2264,8 +2277,13 @@ export class Module {
         }
 
         if (newStmt instanceof ForStatement) {
+            const varAssignStmt = new VarAssignmentStmt("", newStmt);
+            varAssignStmt.lineNumber = newStmt.lineNumber;
+
+            newStmt.loopVar = varAssignStmt;
```

Comment:
Although a lineNumber is assigned here the code itself does not appear in the editor at that line. This is just so that valid references can be calculated correctly as they rely on line numbers to be accurate/work at all.